streaming/vision/convert/ade20k.py

```diff
@@ -0,0 +1,153 @@
+# Copyright 2022 MosaicML Composer authors
+
+"""ADE20K streaming dataset conversion scripts."""
+
+import os
+import random
+from argparse import ArgumentParser, Namespace
+from glob import glob
+from typing import Any, Dict, Iterable, List, Tuple
+
+from tqdm import tqdm
+
+from streaming.base import MDSWriter
+from streaming.vision.convert.base import get_list_arg
+
+
+def parse_args() -> Namespace:
+    """Parse command line arguments.
+
+    Args:
+        Namespace: Command line arguments.
+    """
+    args = ArgumentParser()
+    args.add_argument('--in',
+                      type=str,
+                      default='./datasets/ade20k/',
+                      help='Location of Input dataset. Default: ./datasets/ade20k/')
+    args.add_argument(
+        '--out',
+        type=str,
+        default='./datasets/mds/ade20k/',
+        help='Location to store the compressed dataset. Default: ./datasets/mds/ade20k/')
+    args.add_argument('--splits',
+                      type=str,
+                      default='train,val',
+                      help='Split to use. Default: train,val')
+    args.add_argument('--compression',
+                      type=str,
+                      default='zstd:7',
+                      help='Compression algorithm to use. Default: zstd:7')
+    args.add_argument('--hashes',
+                      type=str,
+                      default='sha1,xxh64',
+                      help='Hashing algorithms to apply to shard files. Default: sha1,xxh64')
+    args.add_argument(
+        '--limit',
+        type=int,
+        default=1 << 25,
```

Comment:
please double check shard size limits: because we intend to train this on 8xA100 with 8 workers/process, number of resulting shards should preferably be comfortably above 64

Reply:
Thanks. Reduced the shard size limit to `1 << 22` to suffice that requirement in https://github.com/mosaicml/streaming/pull/5/commits/e1eaca7428674cb2941960d1ad7c39c7091c1b12